lib/fizzy/execute.cpp

```diff
@@ -196,6 +196,48 @@ Instance instantiate(const Module& module)
     return instance;
 }
 
+namespace
+{
+template <typename DstT, typename SrcT>
+DstT extend(SrcT in)
+{
+    using intermediate_type = typename std::conditional<std::is_signed<SrcT>::value,
+        typename std::make_signed<DstT>::type, DstT>::type;
+
+    const auto x = intermediate_type{in};
+    return static_cast<DstT>(x);
+}
+
+template <typename DstT, typename SrcT>
+inline bool load_from_memory(Instance& instance, Stack<uint64_t>& stack, const uint8_t* immediates)
+{
+    const auto address = static_cast<uint32_t>(stack.pop());
+    // NOTE: alignment is dropped by the parser
+    const auto offset = read<uint32_t>(immediates);
+    if ((address + offset + sizeof(SrcT)) > instance.memory.size())
+        return false;
+
+    const auto ret = load<SrcT>(instance.memory, address + offset);
+    stack.push(extend<DstT, SrcT>(ret));
```

Comment:
```suggestion
    stack.push(extend<DstT>(ret));
```
You can omit `SrcT` because it is going to be deduced.